Foundation/NSDecimal.swift

```diff
@@ -1,155 +1,1837 @@
+        get {
+            switch index {
+            case 0: return _mantissa.0
+            case 1: return _mantissa.1
+            case 2: return _mantissa.2
+            case 3: return _mantissa.3
+            case 4: return _mantissa.4
+            case 5: return _mantissa.5
+            case 6: return _mantissa.6
+            case 7: return _mantissa.7
+            case 8: return _mantissa.8
+            case 9: return _mantissa.9
+            case 10: return _mantissa.10
+            case 11: return _mantissa.11
+            case 12: return _mantissa.12
+            case 13: return _mantissa.13
+            case 14: return _mantissa.14
+            case 15: return _mantissa.15
+            case 16 where _extraWide: return _mantissa.16 // used in integerDivide
+            default: fatalError("Invalid index \(index) for _mantissa")
+            }
+        }
+        set {
+            switch index {
+            case 0: _mantissa.0 = newValue
+            case 1: _mantissa.1 = newValue
+            case 2: _mantissa.2 = newValue
+            case 3: _mantissa.3 = newValue
+            case 4: _mantissa.4 = newValue
+            case 5: _mantissa.5 = newValue
+            case 6: _mantissa.6 = newValue
+            case 7: _mantissa.7 = newValue
+            case 8: _mantissa.8 = newValue
+            case 9: _mantissa.9 = newValue
+            case 10: _mantissa.10 = newValue
+            case 11: _mantissa.11 = newValue
+            case 12: _mantissa.12 = newValue
+            case 13: _mantissa.13 = newValue
+            case 14: _mantissa.14 = newValue
+            case 15: _mantissa.15 = newValue
+            case 16 where _extraWide: _mantissa.16 = newValue
+            default: fatalError("Invalid index \(index) for _mantissa")
+            }
+        }
+    }
+    public func compare(to rhs: WideDecimal) -> ComparisonResult {
+        if self._length > rhs._length {
+            return .orderedDescending
+        }
+        if self._length < rhs._length {
+            return .orderedAscending
+        }
+        for i in (0..<self._length).reversed() {
+            if self[i] > rhs[i] {
+                return .orderedDescending
+            }
+            if self[i] < rhs[i] {
+                return .orderedAscending
+            }
+        }
+        return .orderedSame
+    }
+    public static func <(lhs: WideDecimal, rhs: WideDecimal) -> Bool {
+        return lhs.compare(to:rhs) == .orderedAscending
+    }
+    func toDecimal() -> Decimal {
+        var result = Decimal()
+        result._length = self._length
+        for i in 0..<_length {
+            result[i] = self[i]
+        }
+        return result
+    }
+}
 
-public func NSDecimalMultiplyByPowerOf10(_ result: UnsafeMutablePointer<Decimal>, _ number: UnsafePointer<Decimal>, _ power: Int16, _ roundingMode: NSDecimalNumber.RoundingMode) -> NSDecimalNumber.CalculationError { NSUnimplemented() }
+// == Internal (Swifty) functions ==
```

Comment:
It would be good to make these public API and have the same on the Darwin SDK - thoughts?